### PR TITLE
annotate, delete-setter, delete-subst With Subpackages

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This PR is to modify annotate command to deal with subpackages.

### CODE REVIEW

File: cmd/config/internal/commands/annotate.go [hunk 1]
```diff
@@ -39,32 +44,68 @@ func AnnotateCommand(parent string) *cobra.Command {
...
+	ApiVersion         string
+	Namespace          string
+	Path               string
+	RecurseSubPackages bool
 }
 
 func (r *AnnotateRunner) runE(c *cobra.Command, args []string) error {
 	var input []kio.Reader
 	var output []kio.Writer
+	openAPIFileName, err := ext.OpenAPIFileName()
```
Comment (mortent) [inline cmd/config/internal/commands/annotate.go:60]: We should revisit this function. It is strange that it returns an error.

File: cmd/config/internal/commands/annotate.go [hunk 2]
```diff
@@ -39,32 +44,68 @@ func AnnotateCommand(parent string) *cobra.Command {
...
-		rw := &kio.LocalPackageReadWriter{PackagePath: args[0], NoDeleteFiles: true}
+
+		return handleError(c, kio.Pipeline{
+			Inputs:  input,
+			Filters: []kio.Filter{r},
+			Outputs: output,
+		}.Execute())
+	}
+
+	resourcePackagesPaths, err := pathutil.DirsWithFile(args[0], openAPIFileName, true)
```
Comment (mortent) [inline cmd/config/internal/commands/annotate.go:76]: Is this needed? It seems like just using the existing functionality on the `LocalPackageReader` with the `InludeSubpackages` field should be sufficient?

File: cmd/config/internal/commands/annotate.go (hunk 1 above)
Comment (phanimarupaka) [reply cmd/config/internal/commands/annotate.go:60]: The method uses a method which should be overridden by upstream. This check is to handle any errors in that case.

File: cmd/config/internal/commands/annotate.go (hunk 2 above)
Comment (phanimarupaka) [reply cmd/config/internal/commands/annotate.go:76]: We will get rid of InludeSubpackages filed as a whole. So it is necessary to get all the subpackages paths.

File: cmd/config/internal/commands/annotate.go (hunk 1 above)
Comment (mortent) [reply cmd/config/internal/commands/annotate.go:60]: But in this case the reason we need to account for an error here is that we are using a bit of a hack to get the name. We should revisit the `ext` package to see if we can separate defining the name of the krmfile from the path to the file in the root package.

**Review by mortent — COMMENTED** [review]
Looks good, but can we squash the commits?

**Comment by phanimarupaka** [comment]
@mortent done

**Pull request merged**